index.js

```diff
@@ -89,20 +98,55 @@ class ClientServiceBase {
    * @return {Promise<ClientServiceResponse>}
    */
   async registerCertificate() {
-    // TODO re-implement later
-    // const request = new CertificateRegistrationRequestBuilder(
-    //     new this.protobuf.CertificateRegistrationRequest()
-    // ).withCertHolderId(this.certHolderId)
-    //     .withCertVersion(this.certVersion)
-    //     .withCertPem(this.certPem)
-    //     .build();
+    const request = new CertificateRegistrationRequestBuilder(
+        new this.protobuf.CertificateRegistrationRequest(),
+    ).withCertHolderId(this.certHolderId).
+        withCertVersion(this.certVersion).
+        withCertPem(this.certPem).
+        build();
     return this.sendRequest('registerCert', () =>
-        new Promise((_, reject) => {
-          reject(new Error('This method will be reimplemented later'));
+        new Promise((resolve, reject) => {
+          this.ledgerPrivileged.registerCertificate(request, this.metadata,
+              (err, response) => {
+                if (err) {
+                  reject(err);
+                } else {
+                  resolve(response);
+                }
+              });
         }),
     );
   }
 
+  /**
+   * @return {Promise<ClientServiceResponse>}
+   */
+  async registerFunction(id, name, functionBytes) {
```

Comment:
Could you add the function documentation that describes the param type/ function like it is done on the registerContract method for example ?